tsdb/index/postings.go

```diff
@@ -559,12 +560,14 @@ type mergedPostings struct {
 	err         error
 }
 
-func newMergedPostings(p []Postings) (m *mergedPostings, nonEmpty bool) {
+func newMergedPostings(ctx context.Context, p []Postings) (m *mergedPostings, nonEmpty bool) {
 	ph := make(postingsHeap, 0, len(p))
 
 	for _, it := range p {
 		// NOTE: mergedPostings struct requires the user to issue an initial Next.
 		switch {
+		case ctx.Err() != nil:
+			return &mergedPostings{err: ctx.Err()}, true
```

Comment:
I don't think this check is really necessary - this loop is doing an initial `Next()` on each of the postings, but the main work is done on each subsequent call to `Next()` on the merged structure.

Reply:
I think I agree in retrospect, that you don't gain much here. My thinking was probably that `it.Next()` is potentially slow?